packages/element/src/serialize.js

```diff
@@ -295,13 +306,17 @@ export function escapeLessThan( value ) {
  * "[...] the text cannot contain an ambiguous ampersand [...] must not contain
  * any literal U+0022 QUOTATION MARK characters (")"
  *
+ * Additionally we escape the greater than symbol, as this is used by wptexturize to
```

Comment:
As far as `@wordpress/element` is concerned, it should have no knowledge of a `wptexturize` existing. Further, there's nothing about this module which implies any relationship to being used in the context of a post (i.e. it can be used to produce HTML which would be injected in an admin screen). These modules are meant to be independently useful without consideration being strongly bound to any one specific context. Holding ourselves honest to this guideline will prevent biasing implementation toward our specific needs.

Now, with that said, it may still be our best option moving forward, though I'm also wondering a few things:

- Can this be fixed from `wptexturize`?
- Can this be fixed by running a string replace on the result generated by a serialize call? Or do we need to be granular about the replacement occurring within an attribute value?
- And if so, should we consider extension points to our handling of various parts of the serialization? An attribute value _visitor_, for example?

Considering WordPress' needs as foreign can help surface where these sorts of extension points might be broadly useful.

Reply:
> these modules are meant to be independently useful

I wasn't aware of that, but it's good to know. This did seem like a surely-this-is-too-easy kind of fix and I was expecting it to raise a question or two!

My first thought was to look at `wptexturize`. It's a monster of an old function and seems complicated to fix without triggering consequences somewhere. Saying that, I'm unfamiliar with the process, and don't know if Gutenberg is going to be tied to a particular base WP version etc. A fix there is certainly ideal, but tackling it in Gutenberg seems more immediately practical.

I then looked at fixing it inside the blocks themselves, but this places the burden on the block creator.

I looked at the serialize code. I did consider doing a string replace, but fixing a regex bug with another regex seemed like double trouble! It should be possible though, and we don't need to be granular about it - any unencoded `>` in an attribute will break the post. This could leave the core Gutenberg code to produce clean HTML, and we then have a 'texturizing' layer.

I also looked further into the serialization code and wondered if we could do it at a higher level - somehow traverse the block output and add the extra encoding to attributes. It got complicated, and I then came across `escapeAttribute`, which as solutions go seemed ideally magical.

By extension points do you mean hooks? That would work, and we could target different environments. If `wptexturize` is fixed in a future WordPress, the `>` encoding could be removed etc

Reply:
>By extension points do you mean hooks? That would work, and we could target different environments. If wptexturize is fixed in a future WordPress, the > encoding could be removed etc

In my mind I had pictured something like an AST / other tree visitor, which could be passed as an option to serialize, or in the creation of a custom serializer (e.g. expose serialize factory method to create / override default behavior).

Prior art:

- https://github.com/ctrlplusb/react-tree-walker#visitor
- https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#toc-visitors

Reply:
@WordPress/gutenberg-core I know there is a core ticket for this https://core.trac.wordpress.org/ticket/45387 It doesn't seem like it has progressed much. Shall we land this fix as a preventive measure while the core fix is ongoing?

Reply:
> These modules are meant to be independently useful without consideration being strongly bound to any one specific context.

Using `>` in an attribute is valid HTML. Using `&gt;` in an attribute is also valid HTML. Important to this discussion: the two have the same meaning.

If we're aiming to be as general as possible, we should pick `&gt;` since it's useable in more contexts.

Reply:
> Using `>` in an attribute is valid HTML. Using `&gt;` in an attribute is also valid HTML. Important to this discussion: the two have the same meaning.

I don't disagree, but:

1. The expressed intention of this function was to apply the minimal transformations necessarily to transform an attribute value to be conformant as per the specification of what constitutes a [valid attribute value](https://w3c.github.io/html/syntax.html#elements-attributes).
2. The bug still remains in WordPress that [valid HTML becomes mangled upon save](https://core.trac.wordpress.org/ticket/45387) (whether from Gutenberg or elsewhere), but because there are few who feel comfortable touching the relevant texturize code, the impetus has been placed on integrators to implement workarounds, as we've done here.
3. None of this matters, as the pull request has been approved on the pragmatic consideration that this be solved.

Reply:
You're right. Thank you for your patience and for humoring me :)

I wanted to use this package elsewhere and found I could not because of the details of its implementation. I attempting to understand why, I found this ticket. My comment was meant to be a springboard into that discussion elsewhere, but I instead derailed the conversation. My apologies.

Reply:
@mdawaffe Not at all derailed, certainly not. I think I'd stressed far too much the idea of being "generic" in my original comment, and it was a good prompt to reevaluate what I'd meant in terms of the stated purpose of a module / function.